lib/src/data/repository/movie_remote_repository.dart

```diff
@@ -0,0 +1,108 @@
+import '../../domain/repository/movie_repository_i.dart';
+import '../datasource/mappers/movie_mapper.dart';
+import '../datasource/remote/api_service.dart';
+import '../model/movie_data.dart';
+
+class MovieRemoteRepository implements IMovieRepository {
+  final ApiService apiService;
+
+  MovieRemoteRepository({
+    required this.apiService,
+  });
+
+  @override
+  EitherErrorManager<List<Movie>> loadPopularMovies() async {
+    try {
+      final response = await apiService.getPopularMovies();
+      if (response.statusCode == Constants.statusCodeOk) {
+        return Right(
+          (jsonDecode(response.body)['results'] as List<dynamic>)
+              .map(
+                (data) => MovieMapper.fromDataToEntity(
+                  MovieData.fromJson(data),
+                ),
+              )
+              .toList(),
+        );
+      } else {
+        throw Exception('${response.statusCode} ${response.body}');
+      }
+    } catch (e) {
+      return Left(e.toString());
+    }
+  }
+
+
+  @override
+  EitherErrorManager<List<Movie>> loadNowPlayingMovies() async {
+    try {
+      final response = await apiService.getNowPlayingMovies();
+      if (response.statusCode == Constants.statusCodeOk) {
+        return Right(
+          (jsonDecode(response.body)['results'] as List<dynamic>)
+              .map(
+                (data) => MovieMapper.fromDataToEntity(
+                  MovieData.fromJson(data),
+                ),
+              )
+              .toList(),
+        );
+      } else {
+        throw Exception('${response.statusCode} ${response.body}');
+      }
+    } catch (e) {
+      return Left(e.toString());
+    }
+  }
+
+  @override
+  EitherErrorManager<List<Movie>> loadTopRatedMovies() async {
+    try {
+      final response = await apiService.getTopRatedMovies();
+      if (response.statusCode == Constants.statusCodeOk) {
+        return Right(
+          (jsonDecode(response.body)['results'] as List<dynamic>)
+              .map(
+                (data) => MovieMapper.fromDataToEntity(
+                  MovieData.fromJson(data),
+                ),
+              )
+              .toList(),
+        );
+      } else {
+        throw Exception('${response.statusCode} ${response.body}');
+      }
+    } catch (e) {
+      return Left(e.toString());
+    }
+  }
+
+  @override
+  EitherErrorManager<List<Movie>> loadUpComingMovies() async {
+    try {
+      final response = await apiService.getUpComingMovies();
+      if (response.statusCode == Constants.statusCodeOk) {
+        return Right(
+          (jsonDecode(response.body)['results'] as List<dynamic>)
+              .map(
+                (data) => MovieMapper.fromDataToEntity(
+                  MovieData.fromJson(data),
+                ),
+              )
+              .toList(),
+        );
+      } else {
+        throw Exception('${response.statusCode} ${response.body}');
+      }
+    } catch (e) {
+      return Left(e.toString());
+    }
```

Comment:
there is a lot of duplicated code here, find a way to abstract this method, you are doing the same the only variant is the endpoint